### DISCUSSION TITLE
Not have Section Landing Pages

### DISCUSSION THREAD

**Discussion by 1000camels** [thegetty/quire#899] [Get Help]
I am looking for a way to not have Section Landing Pages, but for them to appear in the TOC and Menu
I do understand I can control whether it part of the flow of pages by setting the output to none.
Ultimately I want to make either the section title in the toc and menu not clickable or have it redirect to the first page.
Is there a way in Quire to do this?
Or if I wanted to try to implement this with .htaccess file, how would I get that file into _site?

**Comment by geealbers** [comment]
@1000camels Originally, this was supposed to work by creating the Markdown page but not giving it a `layout`. In fact, there are three places in the templates (for the list TOC, the grid TOC, and the sidebar menu) where there is still a check to see if the page has a `layout`. If it doesn't it adds the title of the page but without a link, like you're suggesting. (You can find these three places by searching for `const isPage = !!layout`.) The problem is that we later found out that pages with no `layout` threw errors elsewhere, so now all Quire pages are given a `layout` by default. 

The first step in fixing this would be to define a new YAML value to use other than `layout`, add it to Quire's global page data, and then change the logic in those templates accordingly. For example, we could use something like `linked_page: false` on the page in question. We'd need to add `linked_page` to the `eleventyNavigation` data list in `content/_computed/eleventyComputed.js` (`linked_page: data.linked_page`), and then we could change the logic in those TOC templates to `const isPage = linked_page == false ? false : true`. This would generate the TOC with the page title, but no link as intended.

The next step would be to remove the page from the project's pagination so that it doesn't show up in the Back/Next links in the top navbar or in the bottom buttons on the page, and the `linked_page: false` page is skipped over. I'm not sure how to do that, and our developers haven't had a chance to look at this. I think it's maybe in this part of the `eleventyComputed.js` file?:

```js
pagination: ({ collections, page }) => {
    if (!page || !collections.navigation.length) return {}
    const currentPageIndex = collections.navigation
      .findIndex(({ url }) => url === page.url)
    if (currentPageIndex === -1) return {}
    return {
      currentPage: collections.navigation[currentPageIndex],
      currentPageIndex,
      percentProgress: 100 * (currentPageIndex + 1) / collections.navigation.length,
      nextPage: collections.navigation[currentPageIndex + 1],
      previousPage: collections.navigation[currentPageIndex - 1]
    }
  }
```
It seems like we'd want to set the nextPage or previousPage to skip over any pages with `linked_page: false`. Which I think would mean looking at `collections.navigation[currentPageIndex + 1]` seeing if it should be skipped and if so looking at `collections.navigation[currentPageIndex + 2]` and seeing if it should be skipped, etc.... until we find the next page that shouldn't be skipped and setting that as `nextPage`, and then doing the same for `previousPage`.

**Comment by 1000camels** [comment]
Thank you for this.

I started to implement this and ran into this error:

[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] 1. Having trouble writing to "_site/pdf-epub-half-title/index.html" from "./content/pdf-epub-half-title.md" (via EleventyTemplateError)
[11ty] 2. linked_page is not defined (via ReferenceError)

Also, once I tried to use linked_page in an index.md, I started to get the following error:

[quire] INFO	 Figures                        Processing complete
[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] classes.filter is not a function (via TypeError)

I wonder if there is a better approach, to just define an alternative for layout:
Perhaps something like no-page or heading

**Comment by 1000camels** [comment]
I also am thinking about your advice on overriding content. I definition think it would be good to get version numbers into some of these key files, so we can at least know when a file has been customised and is out of date. Systems like WooCommerce do this well.

**Comment by 1000camels** [comment]
I got this to work. I had forgotten to add Layout and Presentation. Still trying to adjust the other templates to exclude it, but at least my pages are throwing an error

**Comment by Erin-Cecele** [comment]
Hi @1000camels were you able to get this working in a satisfactory way?

**Comment by 1000camels** [comment]
Unfortunately not. 

I had to use an exception to match against the title in the menu and toc templates. It was working but seems to have broken so I need to dive deeper into the templates to understand what is happening
